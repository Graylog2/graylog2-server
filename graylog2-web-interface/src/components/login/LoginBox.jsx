import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

import { Col, Row } from 'components/graylog';

const LoginCol = styled(Col)(({ theme }) => css`
  padding: 15px;
  background-color: ${theme.color.global.contentBackground};
  border: 1px solid ${theme.color.gray[80]};
  border-radius: 4px;
  box-shadow: 0 0 21px rgba(0, 0, 0, 0.75);
  margin-top: 120px;
`);

const LoginBox = ({ children }) => {
  return (
    <div className="container">
      <Row>
        <LoginCol md={4} mdOffset={4} xs={6} xsOffset={3}>
          {children}
        </LoginCol>
      </Row>
    </div>
  );
};

LoginBox.propTypes = {
  children: PropTypes.node.isRequired,
};

export default LoginBox;
