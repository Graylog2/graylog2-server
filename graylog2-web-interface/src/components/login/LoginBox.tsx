/*
 * Copyright (C) 2020 Graylog, Inc.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the Server Side Public License, version 1,
 * as published by MongoDB, Inc.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * Server Side Public License for more details.
 *
 * You should have received a copy of the Server Side Public License
 * along with this program. If not, see
 * <http://www.mongodb.com/licensing/server-side-public-license>.
 */
import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

const Wrapper = styled.div(({ theme }) => css`
  background-color: ${theme.colors.global.contentBackground};
  display: flex;
  justify-content: center;
  align-items: center;
  width: 33%;
  padding: 30px;
  max-width: 440px;
  min-width: 330px;
`);

const Container = styled.div`
  width: 100%;
`;

type Props = {
  children: React.ReactNode,
};

const LoginBox = ({ children }: Props) => (
  <Wrapper className="container">
    <Container>
      {children}
    </Container>
  </Wrapper>
);

LoginBox.propTypes = {
  children: PropTypes.node.isRequired,
};

export default LoginBox;
