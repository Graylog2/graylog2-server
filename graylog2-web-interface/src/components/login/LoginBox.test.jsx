import React from 'react';
import { render } from 'wrappedTestingLibrary';

import LoginBox from './LoginBox';

describe('LoginBox', () => {
  it('renders a button after the input if buttonAfter is passed', () => {
    const { container } = render(<LoginBox><div /></LoginBox>);

    expect(container).toMatchSnapshot();
  });
});
