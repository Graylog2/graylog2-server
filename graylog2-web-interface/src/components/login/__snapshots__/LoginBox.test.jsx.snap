// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginBox renders a button after the input if buttonAfter is passed 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="row"
    >
      <div
        class="LoginBox__LoginCol-sc-1uryto7-0 lpbxFw col-md-4 col-md-offset-4 col-xs-6 col-xs-offset-3"
      >
        <div />
      </div>
    </div>
  </div>
</div>
`;
